### PR TITLE
Fake main

### PR DESCRIPTION
**Pull request by nuca01**
saving most of the code

### CODE REVIEW

**Review by vanokvakhadze — APPROVED** [review]

**Review by GiorgiMM16 — APPROVED** [review]
Job well done

**Review by shkippppper — APPROVED** [review]

File: Collaboration2/Collaboration2/Scenes/Search Page/SearchPageViewModel.swift [hunk 1]
```diff
@@ -11,29 +11,74 @@ import Combine
...
+    // MARK: - Helper Functions
     func fetch(with name: String) {
+        print("fetched for searchPageVM name")
         NetworkService.networkService.getData(urlString: url(with: name)) { (result: Result<[City], Error>) in
             switch result {
             case .success(let data):
                 self.cities = data
             case .failure(let error):
+                print("searvhPageVM withString")
+
```
Comment (sandrogelashvili) [inline Collaboration2/Collaboration2/Scenes/Search Page/SearchPageViewModel.swift:39]: we don't need prints here. delete to make code more clear

File: Collaboration2/Collaboration2/Scenes/Search Page/CitiesList.swift [hunk 2]
```diff
@@ -6,20 +6,52 @@
...
+    
+    private func addFavoriteCity(city: City) {
+        if !favoriteCities.contains(where: { favoriteCity in
+            favoriteCity.latitude == city.latitude && favoriteCity.longitude == city.longitude
+        }) {
+            context.insert(city)
+        }
+    }
+    
+    init(list: [City], cityAdded: @escaping () -> Void) {
```
Comment (sandrogelashvili) [inline Collaboration2/Collaboration2/Scenes/Search Page/CitiesList.swift:52]: initialization has to be under properties before body.

File: Collaboration2/Collaboration2/Scenes/Search Page/SearchPageViewModel.swift [hunk 3]
```diff
@@ -11,29 +11,74 @@ import Combine
...
 
 class SearchPageViewModel: ObservableObject {
+    // MARK: - Properties
     @Published var searchQuery: String = ""
-    var isSearching: Bool = false
+    @Published var isSearching: Bool = false
     @Published var cities: [City]?
     private var cancellables = Set<AnyCancellable>()
+    @Published var citiesAndWeathers: [CityAndWeather] = []
+    private var modelContext: ModelContext
```
Comment (sandrogelashvili) [inline Collaboration2/Collaboration2/Scenes/Search Page/SearchPageViewModel.swift:21]: extra line here

File: Collaboration2/Collaboration2/Scenes/Weather Page/WeatherPageView.swift [hunk 4]
```diff
@@ -9,29 +9,80 @@ import SwiftUI
...
+                        
+                        DetailsInfoForCurrentWeather(viewModel: viewModel)
+                        
+                        HourlyWeather(viewModel: viewModel)
+                        
+                        DailyWeatherView(viewModel: viewModel)
+                    }
+                    .padding(EdgeInsets(top: 0, leading: 16, bottom: 45, trailing: 16))
+                }
+
```
Comment (sandrogelashvili) [inline Collaboration2/Collaboration2/Scenes/Weather Page/WeatherPageView.swift:44]: extra line here

**Review by sandrogelashvili — APPROVED** [review]

**Pull request merged**